README.md

```diff
@@ -32,59 +37,224 @@ The main features of NEMU include
+
+```mermaid
+graph TD;
+am["Build a baremetal app with AM"]
+linux["Build a Linux image containing user app"]
+baremetal[/"Image of baremetal app or OS"/]
+run["Run image with NEMU, XS-GEM5, or Xiangshan processor"]
+
+am-->baremetal
+linux-->baremetal
+baremetal-->run
 ```
 
-### Prepare workloads
 
-**Link your bbl or baremetal app to 0x800a0000**
+### Run in standalone to produce checkpoints
 
-We use [gcpt restorer](https://github.com/OpenXiangShan/NEMU/tree/master/resource/gcpt_restore)
-to restore architectural checkpoints, which is a piece of assembly code.
-We put gcpt restorer at 0x80000000 and put bbl or baremetal app at 0x800a0000.
-So users should link bbl or baremetal app to 0x800a0000 instead of 0x80000000.
-Users can build the [bbl here](https://github.com/OpenXiangShan/riscv-pk/commits/noop-take-cpt) which has been linked to 0x0x800a0000.
-A prebuilt bbl linked to 0x800a0000 is also shipped in ready-to-run.
+Because most of the enterprise users and researchers are more interested in running larger workloads,
+like SPECCPU, on XS-GEM5 or Xiangshan processor.
+To reduce the simulation time of detailed simulation, NEMU serves as a checkpoint producer.
+The flow for producing and running checkpoints is as follows.
+The detailed instructions for each step is described in Section [Howto](#howto).
 
-### Take uniform checkpoints and restore checkpoints
+```mermaid
+graph TD;
+linux["Build a Linux image containing NEMU trap app and user app"]
+bin[/"Image containing Linux and app"/]
+profiling["Boot image with NEMU with SimPoint profiling"]
+bbv[/"SimPoint BBV, a .gz file"/]
+cluster["Cluster BBV with SimPoint"]
+points[/"SimPoint sampled points and weights"/]
+take_cpt["Boot image with NEMU to produce checkpoints"]
+checkpoints[/"Checkpoints, several .gz files of memory image"/]
+run["Run checkpoints with XS-GEM5 or Xiangshan processor"]
 
-Take uniform checkpoints (checkpointing every N instructions).
+linux-->bin
+bin-->profiling
+profiling-->bbv
+bbv-->cluster
+cluster-->points
+points-->take_cpt
+take_cpt-->checkpoints
+checkpoints-->run
 
-```shell
-./build/riscv64-nemu-interpreter \
-    --cpt-interval 10000000 -u -b \
-    -D output_top \
-    -C test \
-    -w linux \
-    -r ./resource/gcpt_restore/build/gcpt.bin \
-    --dont-skip-boot\
-    -I 11000000 ./ready-to-run/linux-0xa0000.bin
 ```
 
-Restore checkpoints (example):
-```shell
-./build/riscv64-nemu-interpreter -b\
-    --restore \
-    ./output_top/test/linux/0/_10000003_.gz
+## Howto
+
+### Install dependencies
+
+Because different distributions have different package management tools, the installation commands are different.
+For Ubuntu, users can install the dependencies with the following command:
+```bash
+apt install build-essential man gcc gdb git libreadline-dev libsdl2-dev zstd libzstd-dev
+```
+
+### Use NEMU as reference design
+
+#### Build reference.so
+
+To build NEMU as reference design, run
+``` bash
+make xxx-ref_defconfig
+make menuconfig  # save
```

Comment:
If I remember it correctly, `make menuconfig` is only required at the first time when NEMU is downloaded (before make any_defconfig).

After that, only make xxx_defconfig is needed to safely build NEMU.